api/agent/agent_test.go

```diff
@@ -123,3 +152,85 @@ func TestHandleDesiredState(t *testing.T) {
 	assert.NoError(t, updAgent.HandleDesiredState(testActivityID, 0, dummyDesiredState))
 	<-ch
 }
+
+func TestHandleDesiredStateCommand(t *testing.T) {
+	mockCtr := gomock.NewController(t)
+	defer mockCtr.Finish()
+
+	mockClient := mocks.NewMockUpdateAgentClient(mockCtr)
+	mockUpdateManager := mocks.NewMockUpdateManager(mockCtr)
+
+	updAgent := &updateAgent{
+		client:  mockClient,
+		manager: mockUpdateManager,
+		ctx:     context.Background(),
+	}
+
+	t.Run("test_correct_desired_state_command", func(t *testing.T) {
+		dsCommand := &types.DesiredStateCommand{
+			Command: types.CommandUpdate,
+		}
+		ch := make(chan bool, 1)
+		mockUpdateManager.EXPECT().Command(context.Background(), testActivityID, dummyDesiredStateCommand).DoAndReturn(func(ctx context.Context, activityID string, command *types.DesiredStateCommand) {
+			ch <- true
+			assert.True(t, reflect.DeepEqual(dsCommand, command))
+		})
+		assert.NoError(t, updAgent.HandleDesiredStateCommand(testActivityID, 0, dummyDesiredStateCommand))
+		<-ch
+	})
+}
+
+func TestHandleCurrentStateGet(t *testing.T) {
+	mockCtr := gomock.NewController(t)
+	defer mockCtr.Finish()
+
+	mockClient := mocks.NewMockUpdateAgentClient(mockCtr)
+	mockUpdateManager := mocks.NewMockUpdateManager(mockCtr)
+
+	updAgent := &updateAgent{
+		client:  mockClient,
+		manager: mockUpdateManager,
+		ctx:     context.Background(),
+	}
+
+	inventory := &types.Inventory{
```

Comment:
I've seen the same `inventory` throughout the other tests as well, please consider creating a new file with common test constants